ants/ant.cpp

```diff
@@ -66,59 +78,33 @@ ant step(ant curr, grid landscape, coord spawn_point, int gen_flag){
     left.x = curr.pos.x - 1;    left.y = curr.pos.y;
     right.x = curr.pos.x + 1;   right.y = curr.pos.y;
 
+    coord last = curr.antpath.back();
     // determine possible movement
-    for (int i = 0; i < 4; i++){
```

Comment:
This loop-switch is completely unnecessary. All the cases will be selected consecutively.